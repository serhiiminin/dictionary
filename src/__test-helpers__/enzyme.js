import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const shallowWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()
  return shallow(tree, { context })
}

const mountWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()

  return mount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  })
}

export { shallowWithTheme, mountWithTheme }
