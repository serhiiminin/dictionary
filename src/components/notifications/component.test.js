import React from 'react';
import { shallow } from 'enzyme';
import Notifications from './component';

describe('Notifications', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Notifications
        notifications={[]}
        hideNotification={() => {}}
      >
        children
      </Notifications>
    );
  });
  test('render empty', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('render with one item', () => {
    wrapper.setProps({
      notifications: [{
        id: 'id',
        text: 'text',
        type: 'success',
      }]
    });
    expect(wrapper).toMatchSnapshot();
  })
});
