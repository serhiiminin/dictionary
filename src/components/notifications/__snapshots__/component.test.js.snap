// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications render empty 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Notifications
    classes={Object {}}
    notifications={Array []}
>
    children
</Notifications>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        "children",
        <TransitionGroup
          childFactory={[Function]}
          className={undefined}
          component="ul"
>
          
</TransitionGroup>,
      ],
    },
    "ref": null,
    "rendered": Array [
      "children",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "childFactory": [Function],
          "children": Array [],
          "className": undefined,
          "component": "ul",
        },
        "ref": null,
        "rendered": Array [],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          "children",
          <TransitionGroup
            childFactory={[Function]}
            className={undefined}
            component="ul"
>
            
</TransitionGroup>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "children",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "childFactory": [Function],
            "children": Array [],
            "className": undefined,
            "component": "ul",
          },
          "ref": null,
          "rendered": Array [],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Notifications render with one item 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Notifications
    classes={Object {}}
    notifications={
        Array [
            Object {
              "id": "id",
              "text": "text",
              "type": "success",
            },
          ]
    }
>
    children
</Notifications>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        "children",
        <TransitionGroup
          childFactory={[Function]}
          className={undefined}
          component="ul"
>
          <Transition
                    appear={false}
                    enter={true}
                    exit={true}
                    in={false}
                    mountOnEnter={false}
                    onEnter={[Function]}
                    onEntered={[Function]}
                    onEntering={[Function]}
                    onExit={[Function]}
                    onExited={[Function]}
                    onExiting={[Function]}
                    timeout={300}
                    unmountOnExit={true}
          >
                    [Function]
          </Transition>
</TransitionGroup>,
      ],
    },
    "ref": null,
    "rendered": Array [
      "children",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "childFactory": [Function],
          "children": Array [
            <Transition
              appear={false}
              enter={true}
              exit={true}
              in={false}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={300}
              unmountOnExit={true}
>
              [Function]
</Transition>,
          ],
          "className": undefined,
          "component": "ul",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "id",
            "nodeType": "class",
            "props": Object {
              "appear": false,
              "children": [Function],
              "enter": true,
              "exit": true,
              "in": false,
              "mountOnEnter": false,
              "onEnter": [Function],
              "onEntered": [Function],
              "onEntering": [Function],
              "onExit": [Function],
              "onExited": [Function],
              "onExiting": [Function],
              "timeout": 300,
              "unmountOnExit": true,
            },
            "ref": null,
            "rendered": [Function],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          "children",
          <TransitionGroup
            childFactory={[Function]}
            className={undefined}
            component="ul"
>
            <Transition
                        appear={false}
                        enter={true}
                        exit={true}
                        in={false}
                        mountOnEnter={false}
                        onEnter={[Function]}
                        onEntered={[Function]}
                        onEntering={[Function]}
                        onExit={[Function]}
                        onExited={[Function]}
                        onExiting={[Function]}
                        timeout={300}
                        unmountOnExit={true}
            >
                        [Function]
            </Transition>
</TransitionGroup>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "children",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "childFactory": [Function],
            "children": Array [
              <Transition
                appear={false}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={300}
                unmountOnExit={true}
>
                [Function]
</Transition>,
            ],
            "className": undefined,
            "component": "ul",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "id",
              "nodeType": "class",
              "props": Object {
                "appear": false,
                "children": [Function],
                "enter": true,
                "exit": true,
                "in": false,
                "mountOnEnter": false,
                "onEnter": [Function],
                "onEntered": [Function],
                "onEntering": [Function],
                "onExit": [Function],
                "onExited": [Function],
                "onExiting": [Function],
                "timeout": 300,
                "unmountOnExit": true,
              },
              "ref": null,
              "rendered": [Function],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
